simulations/vip-140/vip-140/simulation.ts

```diff
@@ -0,0 +1,199 @@
+    callbackAfterExecution: async txResponse => {},
+  });
+
+  describe("Post-VIP behavior", async () => {
+    
+    let resilientOracleProxy: ethers.Contract;
+    let chainlinkOracleProxy: ethers.Contract;
+    let boundValidatorProxy: ethers.Contract;
+    let binanceOracleProxy: ethers.Contract;
+    let twapOracleProxy: ethers.Contract;
+    let pythOracleProxy: ethers.Contract;
+    let mockVToken: ethers.Contract;
+    let resilientOracle: ethers.Contract;
+    let comptroller: ethers.Contract;
+    let binanceOracle: ethers.Contract;
+    let timelockSigner: Signer
+
+    before(async () => {
+      await impersonateAccount(PROXY_ADMIN);
+      await impersonateAccount(NORMAL_TIMELOCK);
+      timelockSigner = await ethers.getSigner(NORMAL_TIMELOCK)
+      const signer = await ethers.getSigner(PROXY_ADMIN)
+
+      resilientOracleProxy = new ethers.Contract(RESILIENT_ORACLE, PROXY_ABI, signer);
+      chainlinkOracleProxy = new ethers.Contract(CHAINLINK_ORACLE, PROXY_ABI, signer);
+      boundValidatorProxy = new ethers.Contract(BOUND_VALIDATOR, PROXY_ABI, signer);
+      binanceOracleProxy = new ethers.Contract(BINANCE_ORACLE, PROXY_ABI, signer);
+      twapOracleProxy = new ethers.Contract(TWAP_ORACLE, PROXY_ABI, signer);
+      pythOracleProxy = new ethers.Contract(PYTH_ORACLE, PROXY_ABI, signer);
+
+      comptroller = new ethers.Contract(COMPTROLLER, COMPTROLLER_ABI, provider);
+      resilientOracle = new ethers.Contract(await comptroller.oracle(), RESILIENT_ORACLE_ABI, provider);
+      binanceOracle = new ethers.Contract(BINANCE_ORACLE, BINANCE_ORACLE_ABI, timelockSigner);
+
+      await impersonateAccount(DUMMY_SIGNER);
+      const factory = new ethers.ContractFactory(
+        MOCK_VTOKEN_ABI,
+        MOCK_VTOKEN_CODE,
+        await ethers.getSigner(DUMMY_SIGNER),
+      );
+      mockVToken = await factory.deploy();
+    });
+
+    it("validate implementation address", async () => {
+      expect(await resilientOracleProxy.callStatic.implementation()).to.be.equal(RESILIENT_ORACLE_IMPL)
+      expect(await chainlinkOracleProxy.callStatic.implementation()).to.be.equal(CHAINLINK_ORACLE_IMPL)
+      expect(await boundValidatorProxy.callStatic.implementation()).to.be.equal(BOUND_VALIDATOR_IMPL)
+      expect(await binanceOracleProxy.callStatic.implementation()).to.be.equal(BINANCE_ORACLE_IMPL)
+      expect(await twapOracleProxy.callStatic.implementation()).to.be.equal(TWAP_ORACLE_IMPL)
+      expect(await pythOracleProxy.callStatic.implementation()).to.be.equal(PYTH_ORACLE_IMPL)
+    });
+
+    it("validate IL vToken prices", async () => {
+      for (let i = 0; i < ilPoolTokens.length; i++) {
+        const vToken = ilPoolTokens[i];
+        await binanceOracle.setMaxStalePeriod(vToken.name, 7 * 24 * 60 * 60);
+        await mockVToken.setUnderlyingAsset(vToken.assetAddress);
+        const price = await resilientOracle.getUnderlyingPrice(mockVToken.address);
```

Comment:
I would also call the new `getPrice(asset)`, and I would compare the price is the expected one